### ISSUE TITLE
Pytest conftest.py file update request

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#5679]
_Issue [JP-1881](https://jira.stsci.edu/browse/JP-1881) was created on JIRA by [Alicia Canipe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=acanipe):_

I was hoping to add a plug-in to the [conftest.py file](https://github.com/spacetelescope/jwst/blob/master/jwst/conftest.py) in the jwst repo so that when INS runs the pytests, the docstrings for the tests get printed. This allows more visibility into what is already tested by default, rather than only having the name of the test to describe what is happening. 

Because I wasn't sure how this would affect CI test runs for the software, I wanted to check first before attempting to add the code: 

 
```java
import pytest
import inspect


@pytest.mark.trylast
def pytest_configure(config):
    terminal_reporter = config.pluginmanager.getplugin('terminalreporter')
    config.pluginmanager.register(TestDescriptionPlugin(terminal_reporter), 'testdescription')


class TestDescriptionPlugin:


    def __init__(self, terminal_reporter):
        self.terminal_reporter = terminal_reporter
        self.desc = None


    def pytest_runtest_protocol(self, item):
        self.desc = inspect.getdoc(item.obj)


    @pytest.hookimpl(hookwrapper=True, tryfirst=True)
    def pytest_runtest_logstart(self, nodeid, location):
        if self.terminal_reporter.verbosity == 0:
            yield
        else:
            self.terminal_reporter.write('\n')
            yield
            if self.desc:
                    self.terminal_reporter.write(f'\n{self.desc} ')
```

**Comment by jdavies-st** [comment]
Why don't you try a PR implementing this and we can discuss it a bit in the PR?  I would say that printing docstrings should be turned off by default as well.  I.e. a command line flag might turn it on in your plugin?   Anyway, toss a PR our way.

Another built-in option is:

```
$ pytest jwst/datamodels/tests/test_open.py --collect-only -v
================================================ test session starts =================================================
platform darwin -- Python 3.8.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /Users/jdavies/miniconda3/envs/jwst/bin/python
cachedir: .pytest_cache
rootdir: /Users/jdavies/dev/jwst, configfile: setup.cfg
plugins: asdf-2.7.1, requests-mock-1.8.0, doctestplus-0.8.0, ci-watson-0.5, cov-2.10.1, openfiles-0.5.0
collected 32 items                                                                                                   

<Module jwst/datamodels/tests/test_open.py>
  Test datamodel.open
  <Function test_mirirampmodel_deprecation>
    Test that a deprecated MIRIRampModel can be opened
  <Function test_check_memory_allocation_env[None-None-True]>
    Check environmental control over memory check
  <Function test_check_memory_allocation_env[0.1-None-False]>
    Check environmental control over memory check
  <Function test_check_memory_allocation_env[0.1-1.0-True]>
    Check environmental control over memory check
  <Function test_check_memory_allocation_env[1.0-0.1-False]>
    Check environmental control over memory check
  <Function test_check_memory_allocation_env[None-0.1-False]>
    Check environmental control over memory check
  <Function test_check_memory_allocation[50-1.0-True-True]>
    Check general operation of check_memory_allocation
  <Function test_check_memory_allocation[50-1.0-False-False]>
    Check general operation of check_memory_allocation
  <Function test_check_memory_allocation[100-1.0-True-False]>
    Check general operation of check_memory_allocation
  <Function test_check_memory_allocation[100-None-True-True]>
    Check general operation of check_memory_allocation
  <Function test_check_memory_allocation[50-0.1-True-False]>
    Check general operation of check_memory_allocation
  <Function test_open_from_pathlib>
    Test opening a PurePath object
  <Function test_open_fits>
    Test opening a model from a FITS file
  <Function test_open_fits_s3>
    Test opening a model from a FITS file on S3
  <Function test_open_asdf_s3>
    Test opening a model from an ASDF file on S3
  <Function test_open_association>
    Test for opening an association
  <Function test_container_open_asn_with_sourcecat>
  <Function test_open_none>
  <Function test_open_shape>
  <Function test_open_illegal>
  <Function test_open_hdulist>
  <Function test_open_ramp>
    Open 4D data without a DQ as RampModel
  <Function test_open_cube>
    Open 3D data as CubeModel
  <Function test_open_reffiles[ReferenceFileModel-None]>
    Try opening files with a REFTYPE keyword and different data/dq shapes
  <Function test_open_reffiles[ReferenceImageModel-shape1]>
    Try opening files with a REFTYPE keyword and different data/dq shapes
  <Function test_open_reffiles[ReferenceCubeModel-shape2]>
    Try opening files with a REFTYPE keyword and different data/dq shapes
  <Function test_open_reffiles[ReferenceQuadModel-shape3]>
    Try opening files with a REFTYPE keyword and different data/dq shapes
  <Function test_open_readonly[.asdf]>
    Test opening a FITS-format datamodel that is read-only on disk
  <Function test_open_readonly[.fits]>
    Test opening a FITS-format datamodel that is read-only on disk
  <Function test_open_asdf_no_datamodel_class[asdf]>
  <Function test_open_asdf_no_datamodel_class[fits]>
  <Function test_open_asdf>

============================================ 32 tests collected in 0.21s =============================================
```

Tests with docstrings have the docstrings printed when `-v` is used.

**Comment by stscijgbot-jp** [comment]
_Comment by [Alicia Canipe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=acanipe) on [JIRA](https://jira.stsci.edu/browse/JP-1881?focusedCommentId=506494#comment-506494):_

Thanks [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies), sounds good. I'll give it a try. 

As INS is working on validation notebooks, they said it would be helpful to see what tests are already run. So these pytest runs would happen in our own notebook repo process when we need them to, and the results can be captured on the webpage together with the rest of the notebooks for a complete list of all the tests that were run. Then as INS refines their notebooks, if they end up being pass/fail tests, we can move that code over into the jwst repo, and it would still get captured on our website. But for now this is a convenience to help with our test reporting. Using the plug-in the pytest results print this way, which is a little more helpful for a "report" format:
```java
jwst/superbias/tests/test_bias_sub.py::test_basic_superbias_subtraction 
Check basic superbias subtraction. PASSED                                [ 20%]

jwst/superbias/tests/test_bias_sub.py::test_subarray_correction 
Check that the proper subarray is extracted from the full frame
reference file during subtraction. PASSED                                [ 40%]

jwst/superbias/tests/test_bias_sub.py::test_dq_propagation 
Check that the PIXELDQ array of the science exposure is correctly
combined with the reference file DQ array. PASSED                        [ 60%]

jwst/superbias/tests/test_bias_sub.py::test_nans_in_superbias 
Check that no superbias subtraction is done for pixels that have a
value of NaN in the reference file. PASSED                               [ 80%]

jwst/superbias/tests/test_bias_sub.py::test_full_step 
Test full run of the SuperBiasStep. PASSED                               [100%]

--- generated html file: file:///Users/acanipe/jwst/tmpdir/unit_report.html ----
============================== 5 passed in 26.03s ==============================
```

**Comment by jdavies-st** [comment]
Looks good!  Yeah, definitely want a command line flag such as `--print-docstrings` or similar so this doesn't happen by default.

And also feel free to start updating test docstrings via PR or file an issue to do so for tests missing them, or tests where the docstring is not descriptive enough for your purposes.

**Comment by stscijgbot-jp** [comment]
_Comment by [Alicia Canipe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=acanipe) on [JIRA](https://jira.stsci.edu/browse/JP-1881?focusedCommentId=506507#comment-506507):_

Yes, it's on my (really really long) to-do list, but once I get some of the big stuff out of the way I'll be able to help with that. Also here's the PR: [Update conftest](https://github.com/spacetelescope/jwst/pull/5680)

**Comment by nden** [comment]
Resolved in #5680
